src/code/z_demo.c

```diff
@@ -1232,684 +1404,780 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
-        if (csCtx->unk_1A != 0) {
-            csCtx->unk_18 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+        if (csCtx->camAtReady) {
+            // the frame number set here isn't important, it just signals that the camera data has been applied
+            csCtx->camEyeSplinePointsAppliedFrame = cmd->startFrame;
+
+            if (gUseCutsceneCam) {
                 Play_CameraChangeSetting(play, csCtx->subCamId, CAM_SET_CS_0);
                 Play_ChangeCameraStatus(play, sReturnToCamId, CAM_STAT_WAIT);
                 Play_ChangeCameraStatus(play, csCtx->subCamId, CAM_STAT_ACTIVE);
                 Camera_ResetAnim(Play_GetCamera(play, csCtx->subCamId));
-                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->subCamLookAtPoints,
-                                   csCtx->subCamEyePoints, GET_PLAYER(play), relativeToLink);
+                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->camAtPoints, csCtx->camEyePoints,
+                                   GET_PLAYER(play), relativeToPlayer);
             }
         }
     }
 
     while (shouldContinue) {
-        if (((CutsceneCameraPoint*)cmd)->continueFlag == CS_CMD_STOP) {
-            shouldContinue = 0;
+        if (((CutsceneCameraPoint*)script)->continueFlag == CS_CMD_STOP) {
+            shouldContinue = false;
         }
-        cmd += 0x10;
-        size += 0x10;
+
+        script += sizeof(CutsceneCameraPoint);
+        size += sizeof(CutsceneCameraPoint);
     }
 
     return size;
 }
 
-// Command 0x2 & 0x6: Camera At Points
-s32 Cutscene_Command_CameraLookAtPoints(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 relativeToLink) {
-    s32 shouldContinue = 1;
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_UpdateCamAtSpline(PlayState* play, CutsceneContext* csCtx, u8* script, u8 relativeToPlayer) {
+    s32 shouldContinue = true;
+    CsCmdGeneric* cmd = (CsCmdGeneric*)script;
     s32 size;
 
-    cmd += 8;
-    size = 8;
+    script += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
 
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((D_8015FCC0 < cmdBase->startFrame) || (D_8015FCC0 >= 0xF000))) {
-        csCtx->unk_1A = 1;
-        csCtx->subCamLookAtPoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1B != 0) {
-            D_8015FCC0 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((gCamAtSplinePointsAppliedFrame < cmd->startFrame) || (gCamAtSplinePointsAppliedFrame >= 0xF000))) {
+        csCtx->camAtReady = true;
+        csCtx->camAtPoints = (CutsceneCameraPoint*)script;
+
+        if (csCtx->camEyeReady) {
+            // the frame number set here isn't important, it just signals that the camera data has been applied
+            gCamAtSplinePointsAppliedFrame = cmd->startFrame;
```

Comment:
This looks somewhat important to me

Since for some reason the data for cutscenes has overlapping frame ranges
for example:
```
oot/src/overlays/actors/ovl_Demo_Sa/z_demo_sa_cutscene_data.inc.c
  114,5:     CS_CAM_AT_SPLINE(0, 1270),
  125,5:     CS_CAM_AT_SPLINE(190, 420),
  133,5:     CS_CAM_AT_SPLINE(263, 1383),
  139,5:     CS_CAM_AT_SPLINE(333, 1453),
  145,5:     CS_CAM_AT_SPLINE(403, 1523),
  151,5:     CS_CAM_AT_SPLINE(473, 1745),
  163,5:     CS_CAM_AT_SPLINE(609, 1000),
  173,5:     CS_CAM_AT_SPLINE_REL_TO_PLAYER(685, 1895),
```

The `(csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame)` check is verified for several ranges at the same frame, so this `gCamAtSplinePointsAppliedFrame` logic guarantees the script won't reuse "previous" lists as well as the current one,
it's not just "don't apply the data for the current range twice" like I think the comment hints at

(not that I understand why the scripts have overlapping ranges in the first place... Maybe the end frames are more or less garbage)